### PR TITLE
Slash command for ,lookup

### PR DESCRIPTION
**Pull request by taahamahdi**

### CODE REVIEW

**Review by hoehoetvhss — DISMISSED** [review]

**Review by hoehoetvhss — APPROVED** [review]

File: src/commands/game_commands/lookup.ts [hunk 1]
```diff
@@ -430,12 +570,183 @@ export default class LookupCommand implements BaseCommand {
...
+     * @param interaction - The interaction
+     * @param _messageContext - The message context
+     */
+    async processChatInputInteraction(
+        interaction: CommandInteraction,
+        _messageContext: MessageContext
+    ): Promise<void> {
+        const argument = interaction.data.options[0];
+
+        if (argument.name === "song_link") {
```
Comment (Brainicism) [inline src/commands/game_commands/lookup.ts:598]: Use new interaction helper methods

File: src/commands/game_commands/lookup.ts [hunk 2]
```diff
@@ -430,12 +570,183 @@ export default class LookupCommand implements BaseCommand {
...
+     */
+    static async processAutocompleteInteraction(
+        interaction: Eris.AutocompleteInteraction
+    ): Promise<void> {
+        const autocompleteField = (
+            interaction.data.options[0] as Eris.InteractionDataOptionsSubCommand
+        ).options.filter((x) => x["focused"])[0] as AutocompleteEntry;
+
+        const lowercaseUserInput = autocompleteField.value.toLocaleLowerCase();
+        if (autocompleteField.name === "song_name") {
```
Comment (Brainicism) [inline src/commands/game_commands/lookup.ts:644]: Same here

File: src/commands/game_commands/lookup.ts [hunk 3]
```diff
@@ -430,12 +577,166 @@ export default class LookupCommand implements BaseCommand {
...
+                interaction,
+                interactionData.interactionOptions["song_link"]
+            );
+        } else if (interactionData.interactionName === "song_name") {
+            let songName: string = "";
+            let artistName: string;
+            for (const option of Object.entries(
+                interactionData.interactionOptions
+            )) {
+                if (option[0] === "song_name") {
```
Comment (taahamahdi) [inline src/commands/game_commands/lookup.ts:615]: Do this directly

File: src/commands/game_commands/lookup.ts [hunk 4]
```diff
@@ -430,12 +577,166 @@ export default class LookupCommand implements BaseCommand {
...
+        const focusedKey = interactionData.focusedKey;
+        const focusedVal = interactionData.interactionOptions[focusedKey];
+
+        const lowercaseUserInput = focusedVal.toLocaleLowerCase();
+        const showHangul =
+            containsHangul(lowercaseUserInput) ||
+            State.getGuildLocale(interaction.guildID) === LocaleType.KO;
+
+        if (focusedKey === "song_name") {
+            const artistName = Object.entries(
```
Comment (taahamahdi) [inline src/commands/game_commands/lookup.ts:652]: Directly

**Review by Brainicism — APPROVED** [review]

**Pull request merged**